include/boost/geometry/algorithms/crosses.hpp

```diff
@@ -62,6 +61,82 @@ struct crosses
 {};
 
 
+template <typename Geometry1, typename Geometry2>
+struct crosses<Geometry1, Geometry2, geometry_collection_tag, geometry_collection_tag>
+{
+    template <typename Strategy>
+    static inline bool apply(Geometry1 const& geometry1, Geometry2 const& geometry2,
+                             Strategy const& strategy)
+    {
+        int dimension1 = detail::gc_topological_dimension(geometry1);
+        int dimension2 = detail::gc_topological_dimension(geometry2);
```

Comment:
Minor: can it be const, or even static maybe?

Reply:
`const` yes, `static` no because the dimension depends on what is actually stored in a GC.